### ISSUE TITLE
Logging prefix for call chains on a distributed system

### ISSUE THREAD

**Issue by EliAndrewC** [magfest/sideboard#36]
Moved here from the original (now private) issue tracker.

As a developer trying to debug a problem in production, I'd like a common logging prefix to be shown on all related requests across multiple servers, to help me isolate relevant log messages.

Let's say a user on ``example.com`` makes a websocket request to ``foo.bar``.  Behind the scenes, that method makes a separate request to ``baz.baf``, which is a remote service running on ``example.org``, so we make an websocket RPC request.

When debugging a problem, it can be tricky and annoying to link the requests on different servers to one another.  Since all logging goes through ``sideboard.lib.log``, it would be pretty easy to generate a random logging token and prepend that to our log messages.  When making any outgoing request, we'd add a ``log_token`` field to the websocket JSON, which would be used if present.

**Comment by binary1230** [comment]
YEA, also awesome.  I was thinking about the exact same thing for cherrypy, implementing some kind of request_id for logging

**Comment by EliAndrewC** [comment]
Agreed, I didn't necessarily make this clear in the ticket description, but I intend this as a general feature and not just for websocket call chains.
